installer/bundle/bundle_skel.sh

```diff
@@ -398,61 +398,71 @@ cd $EXTRACT_DIR
 # Do we need to remove the package?
 set +e
 if [ "$installMode" = "R" -o "$installMode" = "P" ]; then
-    pkg_rm omsconfig
-    pkg_rm omsagent
+    if [ -f /opt/microsoft/omsagent/bin/uninstall ]; then
+	/opt/microsoft/omsagent/bin/uninstall $installMode
+    else
+	echo "---------- WARNING WARNING WARNING ----------"
+	echo "Using new shell bundle to remove older kit."
+	echo "Using fallback code to perform kit removal."
+	echo "---------- WARNING WARNING WARNING ----------"
+	echo
 
-    # If MDSD is installed and we're just removing (not purging), leave SCX
+	pkg_rm omsconfig
+	pkg_rm omsagent
 
-    if [ ! -d /var/lib/waagent/Microsoft.OSTCExtensions.LinuxDiagnostic-*/mdsd -o "$installMode" = "P" ]; then
-	if [ -f /opt/microsoft/scx/bin/uninstall ]; then
-	    /opt/microsoft/scx/bin/uninstall $installMode
-	else
-	    for i in /opt/microsoft/*-cimprov; do
-		PKG_NAME=`basename $i`
-		if [ "$PKG_NAME" != "*-cimprov" ]; then
-		    echo "Removing ${PKG_NAME} ..."
-		    pkg_rm ${PKG_NAME}
-		fi
-	    done
+	# If MDSD is installed and we're just removing (not purging), leave SCX
 
-            # Now just simply pkg_rm scx and omi
-	    pkg_rm scx
-	    pkg_rm omi
+	if [ ! -d /var/lib/waagent/Microsoft.OSTCExtensions.LinuxDiagnostic-*/mdsd -o "$installMode" = "P" ]; then
+	    if [ -f /opt/microsoft/scx/bin/uninstall ]; then
+		/opt/microsoft/scx/bin/uninstall $installMode
+	    else
+		for i in /opt/microsoft/*-cimprov; do
+		    PKG_NAME=`basename $i`
+		    if [ "$PKG_NAME" != "*-cimprov" ]; then
```

Comment:
If there are no matching files, then bash will iterate once over `/opt/miccrosoft/*-cimprov`, and I didn't want to pass that to pkg_rm.